.github/workflows/ci.yml

```diff
@@ -0,0 +1,87 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    name: PHP ${{ matrix.php }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php: ['8.0', '8.1']
+        phpunit: ['latest']
+        include:
+        - php: '7.2'
+          phpunit: '8.5.13'
+        - php: '7.3'
+          phpunit: '8.5.13'
+        - php: '7.4'
+          phpunit: '8.5.13'
+          coverage: xdebug
+          flags: '--coverage-clover clover.xml'
+        - php: '8.1'
+          experimental: true
+
+    services:
+      mysql:
+        image: mariadb
+        ports:
+          - 3306/tcp
+        env:
+          MYSQL_ALLOW_EMPTY_PASSWORD: false
+          MYSQL_ROOT_PASSWORD: secret
+          MYSQL_DATABASE: yourls_tests
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Use PHP ${{ matrix.php }}
+      uses: shivammathur/setup-php@2.9.0
+      with:
+        php-version: ${{ matrix.php }}
+        extensions: mbstring, curl, zip, dom, simplexml, intl, pdo_mysql
+        tools: phpunit:${{ matrix.phpunit }}
+        coverage: ${{ matrix.coverage }}
+
+    # - name: Validate composer.json and composer.lock
+    #   run: composer validate
+
+    # - name: Cache Composer packages
+    #   id: composer-cache
+    #   uses: actions/cache@v2
+    #   with:
+    #     path: vendor
+    #     key: ${{ runner.os }}-php-${{ hashFiles('**/composer.lock') }}
+    #     restore-keys: |
+    #       ${{ runner.os }}-php-
+
+    # - name: Install dependencies
+    #   if: steps.composer-cache.outputs.cache-hit != 'true'
+    #   run: composer install --prefer-dist --no-progress --no-suggest
+    #   env:
+    #     APP_ENV: testing
+
+    # - name: Lint
+    #   run: composer run-script lint
```

Comment:
I just don't get why these steps are commented out. 🤔

Reply:
Because we don't use composer as it supposed to be used 😉 
Composer deps are already included, we can't do much on here (except cleaning up these lines if better).